### PR TITLE
Organization refactor

### PR DESCRIPTION
**Pull request by PWAlessi**
Refactored the organization of the repository to align with Golang standards.

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -26,11 +26,11 @@ jobs:
 
     - name: Build
       run: go build -v ./...
-      working-directory: server
+      working-directory: cmd/server
```
Comment (glimchb) [inline .github/workflows/go.yml:29]: can we run build with `.` here ? it will build both client and server... no ?

File: pkg/server/middleend.go [hunk 2]
```diff
@@ -177,4 +183,4 @@ func (s *server) EncryptedVolumeStats(ctx context.Context, in *pb.EncryptedVolum
 	}}, nil
 }
 
-//////////////////////////////////////////////////////////
+// ////////////////////////////////////////////////////////
```
Comment (glimchb) [inline pkg/server/middleend.go:186]: space

**Review by glimchb — COMMENTED** [review]
LGTM, few small comments/questions

File: pkg/server/middleend.go (hunk 2 above)
Comment (PWAlessi) [reply pkg/server/middleend.go]: Fixed

**Review by glimchb — APPROVED** [review]

File: .github/workflows/go.yml (hunk 1 above)
Comment (glimchb) [reply .github/workflows/go.yml:29]: @PWAlessi let's open an issue here https://github.com/opiproject/opi-spdk-bridge/issues so we don't forget to solve it later

**Pull request merged**